subsys/logging/log_output.c

```diff
@@ -287,6 +286,50 @@ static int timestamp_print(const struct log_output *output,
 							"[%5lu.%06d] ",
 #endif
 							total_seconds, ms * 1000U + us);
+			} else if (IS_ENABLED(CONFIG_LOG_OUTPUT_FORMAT_DATE_TIMESTAMP)) {
+#if defined(CONFIG_REQUIRES_FULL_LIBC)
+				char time_str[sizeof("1970-01-01 00:00:00")];
+				struct tm *tm_timestamp;
+				time_t time_seconds = total_seconds;
+
+				tm_timestamp = gmtime(&time_seconds);
```

Comment:
`gmtime` is using shared static `struct tm`. This could cause some concurrency problems if some other part of the system is using `gmtime` as well. So maybe replacing all calls of it with `gmtime_r` would be better?

Otherwise looks good to me.